exporter/datadogexporter/internal/clientutil/retrier.go

```diff
@@ -31,13 +31,15 @@ type Retrier struct {
 	cfg      exporterhelper.RetrySettings
 	logger   *zap.Logger
 	scrubber scrub.Scrubber
+	retryNum int64
```

Comment:
There could be data races when updating `retryNum` on concurrent `DoWithRetries` calls. This will not cause any troubles since we don't really use this value outside of unit tests, but it may lead to noisy data race reports. Should we make this atomic?

Reply:
After thinking more on this, I feel it's safer to just return `retryNum` along with `error`. For every `DoWithRetries` we will have to reset `retryNum` to 0, so it doesn't make too much sense to keep this count across calls. Also the reset operation is not thread-safe.